sdks/python/apache_beam/io/iobase.py

```diff
@@ -820,6 +827,24 @@ def display_data(self):
                                       label='Read Source'),
             'source_dd': self.source}
 
+  def to_runner_api_parameter(self, context):
+    return (urns.READ_TRANSFORM,
+            beam_runner_api_pb2.ReadPayload(
+                source=self.source.to_runner_api(context),
+                is_bounded=beam_runner_api_pb2.BOUNDED
+                if self.source.is_bounded()
+                else beam_runner_api_pb2.UNBOUNDED))
+
+  @staticmethod
+  def from_runner_api_parameter(parameter, context):
+    return Read(BoundedSource.from_runner_api(parameter.source, context))
```

Comment:
Because we don't have an UnboundedSource. We do, however, have a (separate) NativeSource which I just now realized may be an issue. Made a common superclass.